File: python/2367.py
class Solution:
    def arithmeticTriplets(self, nums: List[int], diff: int) -> int:
        c = 0
        for i in range(0, len(nums)):
            for j in range(i, len(nums)):
                for k in range(j, len(nums)):
                    if nums[j] - nums[i] == diff and nums[k] - nums[j] == diff:
                        c += 1
        return c

File: rust/2367.rs
impl Solution {
    pub fn arithmetic_triplets(nums: Vec<i32>, diff: i32) -> i32 {
        let mut count = 0;
        for i in 0..nums.len() {
            for j in i..nums.len() {
                for k in j..nums.len() {
                    if nums[j] - nums[i] == diff && nums[k] - nums[j] == diff {
                        count += 1;
                    }    
                }
            }
        }
        count
    }
}
